MBS-7854 update kotlin 1.3.61 -> 1.3.71

dsvoronin opened the pull request:
https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/

Previous PR's clashed with new Dependencies.kt, and effective version was still 1.3.61

Still an AssertionError problem from last PR #220 

```
> Task :subprojects:android-test:test-app:inferStagingAndroidTestUsageForKeeper FAILED
Exception in thread "main" java.lang.AssertionError
        at com.android.tools.r8.PrintUses$UseCollector.addMethod(PrintUses.java:211)
        at com.android.tools.r8.PrintUses$UseCollector.registerInvokeStatic(PrintUses.java:108)
        at com.android.tools.r8.cf.code.CfInvoke.registerUse(CfInvoke.java:101)
        at com.android.tools.r8.graph.CfCode.registerCodeReferences(CfCode.java:376)
        at com.android.tools.r8.graph.LazyCfCode.registerCodeReferences(LazyCfCode.java:204)
        at com.android.tools.r8.graph.DexEncodedMethod.registerCodeReferences(DexEncodedMethod.java:1070)
        at com.android.tools.r8.PrintUses$UseCollector.registerMethod(PrintUses.java:240)
        at com.android.tools.r8.PrintUses$UseCollector.access$1200(PrintUses.java:77)
        at com.android.tools.r8.PrintUses.lambda$analyze$0(PrintUses.java:340)
        at com.android.tools.r8.graph.DexClass.forEachMethod(DexClass.java:294)
        at com.android.tools.r8.PrintUses.analyze(PrintUses.java:340)
        at com.android.tools.r8.PrintUses.main(PrintUses.java:309)
```

dsvoronin committed:
MBS-7854 update kotlin 1.3.61 -> 1.3.70

https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/

Previous PR's clashed with new Dependencies.kt, and effective version was still 1.3.61

dsvoronin commented:
ok, finally found some clues: 
bumping r8 version to 2.0.46 adds more error info:
`Could not find method Method kotlin.reflect.KClasses.getDefaultType Proto LL kotlin.reflect.KType kotlin.reflect.KClass`

dsvoronin commented:
issues created: 
slackhq/keeper/issues/31
https://issuetracker.google.com/issues/151002499

sergioboi commented on settings.gradle.kts:
```diff
@@ -156,7 +154,7 @@ pluginManagement {
 
                 pluginId.startsWith("org.jetbrains.kotlin.") ->
-                    useVersion(kotlinVersion)
+                    useVersion("1.3.70")
```
Why do we need that hardcode?

dsvoronin replied:
Dependencies.kt is not available in settings.gradle.kts

sergioboi commented on settings.gradle.kts:
```diff
@@ -156,7 +154,7 @@ pluginManagement {
 
                 pluginId.startsWith("org.jetbrains.kotlin.") ->
-                    useVersion(kotlinVersion)
+                    useVersion("1.3.70")
```
Do we have access to Dependencies.kt?

dsvoronin replied:
no afaik

dsvoronin replied:
https://docs.gradle.org/current/userguide/upgrading_version_5.html#classes_from_buildsrc_are_no_longer_visible_to_settings_scripts

sergioboi commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -4,7 +4,7 @@ object Dependencies {
         val sentry = "1.7.23"
         val retrofit = "2.6.2"
-        val kotlin = "1.3.61"
+        val kotlin = "1.3.70"
```
Is our goal to delete this?

dsvoronin replied:
we should try, kotlin plugin provides "platform" data

dsvoronin commented:
https://github.com/JetBrains/kotlin/releases/tag/v1.3.71

dsvoronin committed:
MBS-7854 1.3.71 with single place for version

dsvoronin committed:
MBS-7854 remove usages of mockito 1.x

it was breaking keeper infer task (print usages in fact) because of bundled kotlin-reflect classes

update mockito kotlin to 2.2.0
update mockito to 3.3.3

remove all usages of kotlin-reflect
remove PropertyMatcher.kt, will move to internal project as it is uses kotlin reflect extensively

dsvoronin commented:
it was mockito-kotlin 1.5 with bundled kotlin-reflect classes

sergioboi commented:
> it was mockito-kotlin 1.5 with bundled kotlin-reflect classes

BIATCH

dsvoronin committed:
MBS-7854 remove mockito utils module (not used)

dsvoronin committed:
MBS-7854 remove mockito utils module (not used)

dsvoronin committed:
MBS-7854 fix compile error

sergioboi commented on buildSrc/build.gradle.kts:
```diff
@@ -1,5 +1,5 @@
 plugins {
-    id("org.jetbrains.kotlin.jvm") version "1.3.61"
+    `kotlin-dsl`
```
Does buildSrc see systemProp.kotlinVersion or how does it work?
